src/platformsh.js

```diff
@@ -1,67 +1,447 @@
-let jsonConfig;
-
-try{
-  jsonConfig = require('/run/config.json');
-} catch(err) {
-  // Mocking data in dev mode
-  jsonConfig = {
-    info: {
-      limits: {
-        cpu: 1
-      }
-    }
-  };
-}
+'use strict';
+
 /**
-* Read number of CPUs from environment or fallback to the _private_ configuration property
-* Useful for determining the number of processes to fork.
-*/
-function num_of_cpus() {
-  try {
-    if(process.env['OMP_NUM_THREADS']) {
-      return process.env['OMP_NUM_THREADS'];
-    }
-
-    return Math.ceil(jsonConfig.info.limits.cpu);
-  } catch (err) {
-    throw new Error('Could not get number of cpus');
-  }
+ * Decodes a Platform.sh environment variable.
+ *
+ * @param {string} value
+ *   Base64-encoded JSON (the content of an environment variable).
+ *
+ * @return {any}
+ *   An associative array (if representing a JSON object), or a scalar type.
+ */
+let decode = (value) => {
+    return JSON.parse(Buffer.from(value, 'base64'));
 };
 
-function read_base64_json(varName) {
-  try {
-    return JSON.parse(new Buffer(process.env[varName], 'base64').toString());
-  } catch (err) {
-    throw new Error(`no ${varName} environment variable`);
-  }
-};
+/**
+ * Class representing a Platform.sh environment configuration.
+ */
+class PlatformConfig {
+
+    constructor(env = null, prefix = 'PLATFORM_') {
+        this.environmentVariables = env || process.env;
+        this.envPrefix = prefix;
+
+        // Node doesn't support pre-defined object properties in classes, so
+        // this is mostly for documentation but also to ensure there's always
+        // a legal defined value.
+        this.routesDef = [];
+        this.relationshipsDef = [];
+        this.variablesDef = [];
+        this.applicationDef = [];
+
+        if (this.isValidPlatform()) {
```

Comment:
Generally speaking:

- You want to use the `if (!this.isValidPlatform()) { return; }` pattern
- You want to use `const` over `let` whenever possible